src/WorldStorage/WSSAnvil.cpp

```diff
@@ -1656,11 +1656,43 @@ void cWSSAnvil::LoadEntityFromNBT(cEntityList & a_Entities, const cParsedNBT & a
 
 void cWSSAnvil::LoadBoatFromNBT(cEntityList & a_Entities, const cParsedNBT & a_NBT, int a_TagIdx)
 {
-	std::unique_ptr<cBoat> Boat = cpp14::make_unique<cBoat>(0, 0, 0);
+	std::unique_ptr<cBoat> Boat = cpp14::make_unique<cBoat>(0, 0, 0, cBoat::btOak);
 	if (!LoadEntityBaseFromNBT(*Boat.get(), a_NBT, a_TagIdx))
 	{
 		return;
 	}
+
+	int TypeIdx = a_NBT.FindChildByName(a_TagIdx, "Type");
+	cBoat::eType Type = cBoat::btOak;
+	if (TypeIdx > 0)
+	{
+		if (a_NBT.GetString(TypeIdx) == "oak")
```

Comment:
Get the string into a temporary variable, then compare (to avoid searching for it / creating the `AString` multiple times).